addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -0,0 +1,825 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.squeezebox.handler;
+
+import static org.openhab.binding.squeezebox.SqueezeBoxBindingConstants.SQUEEZEBOXSERVER_THING_TYPE;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
+import java.net.Socket;
+import java.net.URLDecoder;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.squeezebox.config.SqueezeBoxServerConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles connection and event handling to a SqueezeBox Server.
+ * 
+ * @author Markus Wolters
+ * @author Ben Jones
+ * @author Dan Cunningham (OH2 Port)
+ * @since 1.4.0
+ */
+public class SqueezeBoxServerHandler extends BaseBridgeHandler {
+	private Logger logger = LoggerFactory
+			.getLogger(SqueezeBoxServerHandler.class);
+
+	public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections
+			.singleton(SQUEEZEBOXSERVER_THING_TYPE);
+
+	// the value by which the volume is changed by each INCREASE or
+	// DECREASE-Event
+	private static final int VOLUME_CHANGE_SIZE = 5;
+	private static final String NEW_LINE = System.getProperty("line.separator");
+
+	private List<SqueezeBoxPlayerEventListener> squeezeBoxPlayerListeners = new CopyOnWriteArrayList<>();
+	private Map<String, SqueezeBoxPlayer> players = new ConcurrentHashMap<String, SqueezeBoxPlayer>();
+	// client socket and listener thread
+	private Socket clientSocket;
+	private SqueezeServerListener listener;
+
+	private String host;
+
+	private int cliport;
+
+	private int webport;
+
+	public SqueezeBoxServerHandler(Bridge bridge) {
+		super(bridge);
+	}
+
+	@Override
+	public void initialize() {
+		connect();
```

Comment:
This method should return fast. Not sure if connect() might be long-running until a timeout if the server is not reachable? So better schedule it in a job.